### ISSUE TITLE
Errors when visiting EIU admin page

### ISSUE THREAD

**Issue by KolyaKorruptis** [MissAllSunday/emailInactiveUsers#3]
Two reproducible errors that happen every time I visit the EIU admin page. This is with the latest verson of EIU here from GitHub.

in_array() expects parameter 2 to be array, boolean given
File: /Themes/default/Admin.template.php
Line: 903 

``` php
<option value="', $option[0], '"', (($option[0] == $config_var['value'] || (!empty($config_var['multiple']) && in_array($option[0], $config_var['value']))) ? ' selected="selected"' : ''), '>', $option[1], '</option>';
```

unserialize() expects parameter 1 to be string, array given
File: /home/systemshock/web/htdocs/Sources/ManageServer.php
Line: 1877 

``` php
$context['config_vars'][$config_var[1]]['value'] = !empty($context['config_vars'][$config_var[1]]['value']) ? unserialize($context['config_vars'][$config_var[1]]['value']) : array();
```

**Comment by MissAllSunday** [comment]
Yep, its an SMF bug that it was suppose to be fixed with 2.0.9 but the package itself has some errors.

The problem is that any empty or non existing var in that page is treated as a boolean false value but for selectable fields, an empty array needs to be used.

This only happens when you go to the admin page and save the changes without selecting any membergroup, selecting any group automatically converts the var into an array and the error disappears.

A fix for this if you don't have 2.0.9:

find:

``` php


                    $context['config_vars'][$config_var[1]]['value'] = unserialize($context['config_vars'][$config_var[1]]['value']);
```

and replace with

``` php

                    $context['config_vars'][$config_var[1]]['value'] = !empty($context['config_vars'][$config_var[1]]['value']) ? unserialize($context['config_vars'][$config_var[1]]['value']) : array();

```

**Comment by MissAllSunday** [comment]
Oh, I see you already have the code. This is a special case then and I'm afraid users who installed this mod prior SMF 2.0.9 will have the same issue :sob:

There are 2 options here, one is to empty the eiu_groups setting from your settings table. 

The other one is to add a is_string() check in addition to the !empty($context['config_vars'][$config_var[1]]['value'])

!empty($context['config_vars'][$config_var[1]]['value']) && is_string($context['config_vars'][$config_var[1]]['value'])

**Comment by KolyaKorruptis** [comment]
I emptied the eiu_groups value from the smf_settings table. That solved the problem as long as I didn't select any groups in the mod's admin panel. Of course I need to select groups. But doing so reproduces the same errors again.

It's not clear to me where is_string() should be added? In which file?

**Comment by MissAllSunday** [comment]
sorry, I forgot about this :(

I'm not sure why the error would re-appear though, gotta investigate that.

the is_string should be added like this:

!empty($context['config_vars'][$config_var[1]]['value']) && is_string($context['config_vars'][$config_var[1]]['value'])

but thats only a semi fix for the unserialized() function, I still don't know why that var has already been converted to an array before been passed to unserialize()
